File: LeetCode/70_ClimbingStairs.cs
public class Solution {
    public int ClimbStairs(int n) {
        if(n == 0)
            return 0;
        if(n == 1)
            return 1;
        if(n == 2)
            return 2;
        int x = 1, y = 2;
        
        int res = 0;
        for(int i=3; i<=n; i++) {
            res = x + y;
            x = y;
            y = res;
        }
        return res;
    }
}

File: LeetCode/70_ClimbingStairs.cpp
class Solution {
    
    long climbStairs(int n, unordered_map<int, long>& map) {
        
       if(n == 0)
            return 1;
        
        if(n < 0)
            return 0;
     
        auto search = map.find(n);
        if(search != map.end())
            return search->second;
        
        long ans = climbStairs(n-1, map) + climbStairs(n-2, map);
        map[n] = ans;
        return ans;
    }
    
public:

    long climbStairs(int n) {
        unordered_map<int, long> map;
        return climbStairs(n, map);

    }
};